src/Blockcore.Indexer.Cirrus/Storage/Mongo/SmartContracts/NFTComputedService.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Blockcore.Indexer.Cirrus.Client;
+using Blockcore.Indexer.Cirrus.Storage.Mongo.Types;
+using Blockcore.Indexer.Core.Client;
+using Blockcore.Indexer.Core.Operations.Types;
+using Microsoft.Extensions.Logging;
+using MongoDB.Driver;
+using MongoDB.Driver.Linq;
+
+namespace Blockcore.Indexer.Cirrus.Storage.Mongo.SmartContracts;
+
+public class NftComputationService : IComputeSmartContractService<NonFungibleTokenComputedTable>
+{
+   readonly ILogger<NftComputationService> logger;
+   readonly ICirrusMongoDb mongoDb;
+   readonly ISmartContractHandlersFactory<NonFungibleTokenComputedTable> logReaderFactory;
+   readonly CirrusClient cirrusClient;
+
+   public NftComputationService(ILogger<NftComputationService> logger,
+      ICirrusMongoDb mongoDb,
+      ISmartContractHandlersFactory<NonFungibleTokenComputedTable> logReaderFactory,
+      ICryptoClientFactory clientFactory,
+      SyncConnection connection)
+   {
+      this.logger = logger;
+      this.mongoDb = mongoDb;
+      this.logReaderFactory = logReaderFactory;
+      cirrusClient = (CirrusClient)clientFactory.Create(connection);
+   }
+
+   public async Task<NonFungibleTokenComputedTable> ComputeSmartContractForAddressAsync(string address)
+   {
+      var contract = await LookupSmartContractForAddressAsync(address);
+
+      if (contract is null)
+         return null;
+
+      var contractTransactions = await mongoDb.CirrusContractTable
+         .AsQueryable()
+         .Where(_ => (_.ToAddress == address ||
```

Comment:
Should we not also filter by `CodeType`? or actually a smart contract address is unique right?